File: symmetric_key_cryptography/ruby/substitution_cipher.rb
require('byebug');

def create_object_key 
  key = {}

  used_indicies = [];
  alpha = 'abcdefghijklmnopqrstuvwxyz'

  (0...alpha.length).each do |i|
    random = 0

    unique = false
    while(!unique) do
        random = rand(0...alpha.length)

        unique = true if(!used_indicies.include?(random))
    end

    used_indicies << random
    key[alpha.split('')[i]] = alpha[random];
  end

  puts key
  key
end

def substitution_cipher (str, key)
  new_str = [];

  str.each_char do |char|
    new_str << key[char]
  end

  encrypted_str = new_str.join('')
  puts encrypted_str
  encrypted_str
end

key1 = create_object_key()
substitution_cipher('abc', key1)

key2 = create_object_key()
substitution_cipher('message', key2)

key3 = create_object_key()
substitution_cipher('substitutionmessage', key3)

File: symmetric_key_cryptography/python/substitution_cipher.py
from random import seed
from random import randrange

def create_object_key():
    key = {}

    used_indices = []
    alpha = 'abcdefghijklmnopqrstuvwxyz'

    for i in range(len(alpha)):
        random_num = 0
        unique = False

        while not unique:
            random_num = randrange(len(alpha))

            if random_num not in used_indices:
                unique = True
        
        # check below lines for syntax
        used_indices.append(random_num)
        key[alpha[i]] = alpha[random_num]

    print(key)
    return key

def substitution_cipher(str, key):
    new_str = []

    for char in str:
        new_str.append(key[char])
    
    encrypted_str = ''.join(new_str)

    print(encrypted_str)
    encrypted_str


key1 = create_object_key()
substitution_cipher('abc', key1)

key2 = create_object_key()
substitution_cipher('message', key2)

key3 = create_object_key()
substitution_cipher('substitutionmessage', key3)

            

